workshops/running-amazon-ec2-workloads-at-scale/spot-interruption-handler.yaml

```diff
@@ -0,0 +1,109 @@
+AWSTemplateFormatVersion: '2010-09-09'
+
+Resources:
+  SpotInterruptionHandlerFunction:
+    Type: AWS::Lambda::Function
+    DependsOn:
+    - LambdaFunctionRole
+    Properties:
+      Handler: index.handler
+      Role: !GetAtt LambdaFunctionRole.Arn
+      Code:
+        ZipFile: !Sub |
+          import boto3
+
+          def handler(event, context):
+          
+              instanceId = event['detail']['instance-id']
+              instanceAction = event['detail']['instance-action']
+              
+              print("Handling spot instance interruption notification for instance {id}".format(id=instanceId))
+              
+              # Describe tags for the instance that will be interrupted
+              try:
+                  ec2client = boto3.client('ec2')
+                  describeTags = ec2client.describe_tags(Filters=[{'Name': 'resource-id','Values':[instanceId]}])
+              except Exception as e:
+                  print("No action being taken. Unable to describe tags for instance id:", instanceId)
+                  print(e)
+                  return
```

Comment:
This is an exception, the exception should be re-raised not swallowed. The exception is properly captured and logged but, we want to notify the framework that something went wrong. This should raise
```
              except Exception as e:
                  print("No action being taken. Unable to describe tags for instance id:", instanceId)
                  print(e)
                  raise e
```

Reply:
Yeah, good catch. Fixed this. Also changed describe_tags for describe_instances to describe the actual instance itself.